contracts/vortex/CarbonVortex.sol

```diff
@@ -631,6 +631,10 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
         if (sourceAmount == 0) {
             revert InvalidTrade();
         }
+        // revert if unnecessary native token is received
+        if (_targetToken != NATIVE_TOKEN && msg.value > 0) {
+            revert UnnecessaryNativeTokenReceived();
+        }
```

Comment:
Can turn this into an if/else construct, similar to the one in the other function, i.e.:
```
        if (_targetToken == NATIVE_TOKEN) {
            if (msg.value < sourceAmount) {
                revert InsufficientNativeTokenSent();
            }
        } else {
            if (msg.value > 0) {
                revert UnnecessaryNativeTokenReceived();
            }
        }
```

Reply:
prefer to keep it as it is